harp/reader.py

```diff
@@ -44,13 +45,37 @@ def __init__(
         self.read = read
 
 
+class RegisterMap(UserDict[str, RegisterReader]):
+    _address_map: dict[int, str]
+
+    def __init__(self, registers: dict[str, RegisterReader]) -> None:
+        super().__init__(registers)
+        self._address_map = {self[key].register.address: key for key in self.keys()}
+
+    def __getitem__(self, __key: Union[str, int]) -> RegisterReader:
+        if isinstance(__key, int):
+            super().__getitem__(self._address_map[__key])
```

Comment:
I think we need to return here and also we can use the `_address_map` dict directly from `self` rather than the base implementation:

```suggestion
            return self._address_map[__key]
```